### ISSUE TITLE
Is the tags count correct?

### ISSUE THREAD

**Issue by albertocottica** [opencarecc/graph-ryder-dashboard#7]
I see @jason-vallet has added a list of stats as per #3 . However, I am worried that the number of tags may not be correct. I have my own script that returns a very different number: 3,915 annotations (not computed by the dashboard) with 867 codes (dashboard says 1,614). 

There are also small differences in the numbers of comments (my script says 1,688, GraphRyders says 1,633) and users (my script says  posts (my script says 316, GraphRyders says 318) The number of posts is the same:  352. 

My script for annotations:

```python
import requests

checklistAnnotations = []
checklistCodes = []

response = requests.get('https://edgeryders.eu/opencare/annotations')
annotations = response.json()

## annotations = requests.get('https://edgeryders.eu/opencare/annotations').json()

latest = ''

for item in annotations['nodes']:
        if item ['node'] ['user_name'] == 'Amelia':
                created = item ['node']['date_standardized']
                if created > latest:
                        latest = created
                if item ['node']['annotation_id'] not in checklistAnnotations:
                        checklistAnnotations.append(item ['node']['annotation_id'])
                if item ['node']['tag_id'] not in checklistCodes:
                        checklistCodes.append(item ['node']['tag_id'])

print ('annotations: ' + str(len(checklistAnnotations)) )
print ('latest on: ' + latest)
print ('codes: ' + str(len(checklistCodes)) )
```

The logic is that MySQL returns a lot of duplicates, and even distinct does not always work. So I need to build these checklists of entities I already encountered, and for each entity returned by the view check that it is not already there.

**Comment by jason-vallet** [comment]
I have used a simple (to not say dumb) request to get the number of tags (and users) and did not checked if they were used, so 1414 is actually the total number of tags existing within the db (as imported from ER), not just the ones used. I will correct this shortly.

I can explain the difference in the number of comments as some appears to not have existing parent (post or comment) when imported. Those elements are thus ignored to avoid discrepancy in the database. 

Do you want to have the number of annotations available with the stats?

**Comment by albertocottica** [comment]
Re tags: OK, that explains it.

Re number of comments: can you make an example (need the number of comment id)? 

Re annotations: yes please.

**Comment by jason-vallet** [comment]
The comments all seem to be pretty old (~5years from what I can tell) so I am guessing their parents are not considered as elements of the Opencare project.

Here are all the problematic ones: 
[comment 3085](https://edgeryders.eu/comment/3085#comment-3085)
[comment 3627](https://edgeryders.eu/comment/3627#comment-3627)
[comment 3574](https://edgeryders.eu/comment/3574#comment-3574)
[comment 3008](https://edgeryders.eu/comment/3008#comment-3008)
[comment 3555](https://edgeryders.eu/comment/3555#comment-3555)
[comment 3541](https://edgeryders.eu/comment/3541#comment-3541)
[comment 3552](https://edgeryders.eu/comment/3552#comment-3552)
[comment 3536](https://edgeryders.eu/comment/3536#comment-3536)
[comment 3538](https://edgeryders.eu/comment/3538#comment-3538)
[comment 3535](https://edgeryders.eu/comment/3535#comment-3535)
[comment 3533](https://edgeryders.eu/comment/3533#comment-3533)
[comment 3532](https://edgeryders.eu/comment/3532#comment-3532)
[comment 3333](https://edgeryders.eu/comment/3333#comment-3333)
[comment 3175](https://edgeryders.eu/comment/3175#comment-3175)
[comment 3184](https://edgeryders.eu/comment/3184#comment-3184)
[comment 3176](https://edgeryders.eu/comment/3176#comment-3176)
[comment 3181](https://edgeryders.eu/comment/3181#comment-3181)
[comment 3158](https://edgeryders.eu/comment/3158#comment-3158)
[comment 3150](https://edgeryders.eu/comment/3150#comment-3150)
[comment 3156](https://edgeryders.eu/comment/3156#comment-3156)
[comment 3153](https://edgeryders.eu/comment/3153#comment-3153)
[comment 3146](https://edgeryders.eu/comment/3146#comment-3146)
[comment 3139](https://edgeryders.eu/comment/3139#comment-3139)
[comment 3068](https://edgeryders.eu/comment/3068#comment-3068)
[comment 2950](https://edgeryders.eu/comment/2950#comment-2950)
[comment 3042](https://edgeryders.eu/comment/3042#comment-3042)
[comment 2922](https://edgeryders.eu/comment/2922#comment-2922)
[comment 2836](https://edgeryders.eu/comment/2836#comment-2836)
[comment 2804](https://edgeryders.eu/comment/2804#comment-2804)
[comment 2461](https://edgeryders.eu/comment/2461#comment-2461)
[comment 2339](https://edgeryders.eu/comment/2339#comment-2339)
[comment 2199](https://edgeryders.eu/comment/2199#comment-2199)
[comment 2277](https://edgeryders.eu/comment/2277#comment-2277)
[comment 2194](https://edgeryders.eu/comment/2194#comment-2194)
[comment 2173](https://edgeryders.eu/comment/2173#comment-2173)
[comment 2150](https://edgeryders.eu/comment/2150#comment-2150)
[comment 2140](https://edgeryders.eu/comment/2140#comment-2140)
[comment 2043](https://edgeryders.eu/comment/2043#comment-2043)
[comment 2037](https://edgeryders.eu/comment/2037#comment-2037)
[comment 1497](https://edgeryders.eu/comment/1497#comment-1497)
[comment 1495](https://edgeryders.eu/comment/1495#comment-1495)
[comment 1490](https://edgeryders.eu/comment/1490#comment-1490)
[comment 1463](https://edgeryders.eu/comment/1463#comment-1463)
[comment 1468](https://edgeryders.eu/comment/1468#comment-1468)
[comment 1433](https://edgeryders.eu/comment/1433#comment-1433)
[comment 1449](https://edgeryders.eu/comment/1449#comment-1449)
[comment 1418](https://edgeryders.eu/comment/1418#comment-1418)
[comment 1415](https://edgeryders.eu/comment/1415#comment-1415)
[comment 3204](https://edgeryders.eu/comment/3204#comment-3204)
[comment 3205](https://edgeryders.eu/comment/3205#comment-3205)

**Comment by albertocottica** [comment]
Got it. So:

* The sub-comments of these comments are legitimate. We reused some old (but relevant) content in order to seed the conversation.
* They do have their own comment_ids and post_ids, but they also have parent_comment_ids that do not correspond to anything in the database.
* This happened when we migrated the first Edgeryders website (Drupal 6) onto the present one (Drupal 7), in spring 2013. It does not happen anywhere else in the db: it's a one-off.
* It COULD happen again in the future. Under Edgeryders TOCs, each user owns their content. Suppose I comment your comment, and then you delete your comment: what happens then? Drupal has a solution: it displays my comment (previously a comment to your comment) as a comment to the post that originated the thread. In terms of the social interaction network, the edge is redirected: used to be from me to you, but now it's from me to the author of the post.
* Still, I think these comments should be counted. They have annotations, so they induce edges in the co-occurrences graph, etc. 
* They are 50 out of the 53 missing. The remaining 3 can be explained away with the time-delay between your harvesting into the graph db and the Drupal db I am looking into. Good!
